### PR TITLE
[main blog] Reducing support for 32-bit Apple targets

### PR DESCRIPTION
**Pull request by pietroalbini**
[RFC 2837], "Demote Apple 32bit targets to Tier 3", is going to be merged soon, and one of the things it requires is a blog post on the main blog announcing the change, to bring as much awareness to the demotion as we can.

I wrote a draft of the post, what do y'all think?

r? @rust-lang/core
cc @rust-lang/release  

[RFC 2837]: https://github.com/rust-lang/rfcs/pull/2837

### CODE REVIEW

**Comment by pietroalbini** [comment]
The RFC has been merged.

**Comment by pietroalbini** [comment]
Added a section in the end on how the change will affect nightly users.

File: posts/2019-12-30-reducing-support-for-apple-32bit-targets.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+author: Pietro Albini
+---
+
+The Rust team is sad to announce that Rust 1.41.0 (to be released on January
+30th, 2020) will be the last release with the current level of support for
+Apple 32bit targets. Starting from Rust 1.42.0, those targets will be demoted
+to Tier 3.
+
+The decision was made on [RFC 2837], and was accepted by the compiler and
+release teams. This post explains what the change means, why we did it and how
```
Comment (Centril) [inline posts/2019-12-30-reducing-support-for-apple-32bit-targets.md]:
```suggestion
release teams. This post explains what the change means, why we did it, and how
```

File: posts/2019-12-30-reducing-support-for-apple-32bit-targets.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+- Tier 2 targets can also be downloaded through rustup, but our
+  automated builds don’t execute the test suite for them. While we guarantee a
+  compiler build will be available, we don’t ensure it will actually work
+  without bugs (or even work at all).
+
+- Tier 3 targets are not available for download through rustup, and are
+  ignored during our automated builds. You can still build their standard
+  library for cross-compiling (or the full compiler in some cases) from source
+  on your own, but you might encounter build errors, bugs or missing features.
```
Comment (Centril) [inline posts/2019-12-30-reducing-support-for-apple-32bit-targets.md]:
```suggestion
  on your own, but you might encounter build errors, bugs, or missing features.
```

File: posts/2019-12-30-reducing-support-for-apple-32bit-targets.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+targets.
+
+# Why are those targets being demoted?
+
+Apple dropped support for running 32bit binaries starting from [macOS
+10.15][deprecate-macos] and [iOS 11][deprecate-ios]. They also prevented all
+developers from cross-compiling 32bit programs and apps starting from Xcode 10
+(the platform’s IDE, containing the SDKs).
+
+Due to those decisions from Apple, the targets are not much useful to our users
```
Comment (Centril) [inline posts/2019-12-30-reducing-support-for-apple-32bit-targets.md]:
```suggestion
Due to those decisions from Apple, the targets are no longer useful to our users,
```

File: posts/2019-12-30-reducing-support-for-apple-32bit-targets.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+anymore, and their choice to prevent cross-compiling makes it hard for the
```
Comment (Centril) [inline posts/2019-12-30-reducing-support-for-apple-32bit-targets.md]:
```suggestion
and their choice to prevent cross-compiling makes it hard for the
```

**Comment by pietroalbini** [comment]
Good point @jonas-schievink, amended the blog post.

File: posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+title: "Reducing support for 32-bit Apple targets"
+author: Pietro Albini
+---
+
+The Rust team is sad to announce that Rust 1.41.0 (to be released on January
```
Comment (BatmanAoD) [inline posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: I don't think we need to make an emotional statement. Either of these would be fine, I think:

Comment (BatmanAoD) [reply posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]:
```suggestion
The Rust team regrets to announce that Rust 1.41.0 (to be released on January
```

Comment (BatmanAoD) [reply posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: Or, with no "padding" at all:

```suggestion
Rust 1.41.0 (to be released on January
```

File: posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+- Tier 2 targets can also be downloaded through rustup, but our
+  automated builds don’t execute the test suite for them. While we guarantee a
+  compiler build will be available, we don’t ensure it will actually work
+  without bugs (or even work at all).
+
+- Tier 3 targets are not available for download through rustup, and are
+  ignored during our automated builds. You can still build their standard
+  library for cross-compiling (or the full compiler in some cases) from source
+  on your own, but you might encounter build errors, bugs, or missing features.
```
Comment (BatmanAoD) [inline posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: I'm not sure if we need a full summary of how support tiers work in this announcement. I think the links to the current support tiers and RFC 2803, together with a simple statement that 32-bit Apple targets will no longer be exercised in any way via the Rust project's CI infrastructure, would be sufficient.

Comment (pietroalbini) [reply posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: As this affects both tier 1 and tier 2 platforms going into tier 3, I think having a brief description of what those tiers mean in the blog post is useful.

File: posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+being formalized and revised in [RFC 2803]), explaining what we guarantee:
+
+- Tier 1 targets can be downloaded through rustup and are fully tested
+  during the project’s automated builds. A bug or a regression affecting one of
+  these targets is usually prioritized more than bugs only affecting platforms
+  in other tiers.
+
+- Tier 2 targets can also be downloaded through rustup, but our
+  automated builds don’t execute the test suite for them. While we guarantee a
+  compiler build will be available, we don’t ensure it will actually work
```
Comment (Manishearth) [inline posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: nit: some tier 2 targets are std only and we do not guarantee a compiler build

Comment (pietroalbini) [reply posts/2019-12-30-reducing-support-for-32-bit-apple-targets.md]: Clarified this.

**Pull request merged**